--- a/migrate_all_nis_offline.sh	2020-07-10 21:52:06.000000000 +0200
+++ b/migrate_all_nis_offline.sh	2026-08-02 22:30:51.284904438 +0200
@@ -70,7 +70,7 @@
 if [ "$MIGRATION_TYPE" != 'min' ]; then
 	ypcat $DOMFLAG services > $ETC_SERVICES
 	ypcat $DOMFLAG protocols > $ETC_PROTOCOLS
-	ypcat $DOMFLAG rpc.byname > $ETC_RPC
+	ypcat $DOMFLAG rpc.byname | sort | uniq > $ETC_RPC
 	ypcat $DOMFLAG hosts > $ETC_HOSTS
 	ypcat $DOMFLAG networks > $ETC_NETWORKS
 	#ypcat $DOMFLAG -k aliases > $ETC_ALIASES
--- a/migrate_all_nis_online.sh	2020-07-10 21:52:06.000000000 +0200
+++ b/migrate_all_nis_online.sh	2026-08-02 22:30:51.284862608 +0200
@@ -56,7 +56,7 @@
 # Get arguments
 on_getopts $@
 # Generate temp folder
-tmpfd_gen
+tmpfd_gen shadow
 
 question="Enter the NIS domain to import from (optional): "
 echo "$question " | tr -d '\012' > /dev/tty
@@ -68,11 +68,12 @@
 fi
 
 ypcat $DOMFLAG passwd > $ETC_PASSWD
+ypcat $DOMFLAG shadow.byname > $ETC_SHADOW
 ypcat $DOMFLAG group > $ETC_GROUP
 if [ "$MIGRATION_TYPE" != 'min' ]; then
 	ypcat $DOMFLAG services > $ETC_SERVICES
 	ypcat $DOMFLAG protocols > $ETC_PROTOCOLS
-	ypcat $DOMFLAG rpc.byname > $ETC_RPC
+	ypcat $DOMFLAG rpc.byname | sort | uniq > $ETC_RPC
 	ypcat $DOMFLAG hosts > $ETC_HOSTS
 	ypcat $DOMFLAG networks > $ETC_NETWORKS
 	#ypcat $DOMFLAG -k aliases > $ETC_ALIASES
@@ -81,4 +82,4 @@
 # Call main functions
 on_main
 # Destroy temp folder
-tmpfd_destroy
+tmpfd_destroy shadow
--- a/migrate_automount.pl	2020-07-10 21:52:06.000000000 +0200
+++ b/migrate_automount.pl	2026-08-02 22:31:12.159764349 +0200
@@ -84,13 +84,12 @@
 	
 	return if (!$mapname);
 
-	if ($key eq "*"){	
-		# since * isnt a valid attrib, replace it with "/" 
-		# which isnt a valid filename :->
-		print $HANDLE "dn: cn=/,nisMapName=$mapname,$NAMINGCONTEXT\n";
-	} else {
-		print $HANDLE "dn: cn=$key,nisMapName=$mapname,$NAMINGCONTEXT\n";
+	if ($key eq "*"){
+		# since * isn't a valid attrib, replace it with "/" 
+		# which isn't a valid filename :->
+		$key = "/";
 	}
+	print $HANDLE "dn: cn=$key, ou=$mapname,$NAMINGCONTEXT\n";
 	print $HANDLE "objectClass: nisObject\n";
 	print $HANDLE "cn: $key\n";
 	print $HANDLE "nisMapEntry: $value\n";
--- a/migrate_common.sh	2026-08-02 22:30:51.261390534 +0200
+++ b/migrate_common.sh	2026-08-02 22:30:51.284734636 +0200
@@ -174,6 +174,10 @@
 	if [ "X$ETC_PASSWD" = "X" ]; then
 		ETC_PASSWD=/etc/passwd
 	fi
+	if [ "X$ETC_SHADOW" = "X" ]; then
+		ETC_SHADOW=/etc/shadow
+	fi
+	export ETC_SHADOW
 }
 
 setvars_full () {
@@ -502,6 +506,7 @@
 		exit 1
 	}
 	ETC_PASSWD="$(mktemp "$TEMP_DIR/passwd.ldap.XXXXXX")"
+	[ ! -z "$(echo "$@" | grep 'shadow')" ] && ETC_SHADOW="$(mktemp "$TEMP_DIR/shadow.ldap.XXXXXX")"
 	ETC_GROUP="$(mktemp "$TEMP_DIR/group.ldap.XXXXXX")"
 	if [ "$MIGRATION_TYPE" != 'min' ]; then
 		ETC_SERVICES="$(mktemp "$TEMP_DIR/services.ldap.XXXXXX")"
@@ -518,6 +523,7 @@
 
 tmpfd_destroy () {
 	rm -f $ETC_PASSWD
+	[ ! -z "$(echo "$@" | grep 'shadow')" ] && rm -f $ETC_SHADOW
 	rm -f $ETC_GROUP
 	if [ "$MIGRATION_TYPE" != 'min' ]; then
 		rm -f $ETC_SERVICES
--- a/migrate_passwd.pl	2020-07-10 21:52:06.000000000 +0200
+++ b/migrate_passwd.pl	2026-08-02 22:30:51.284806747 +0200
@@ -179,7 +179,7 @@
 
 sub read_shadow_file
 {
-	open(SHADOW, "/etc/shadow") || return;
+	open(SHADOW, defined($ENV{ETC_SHADOW}) ? $ENV{ETC_SHADOW} : "/etc/shadow") || return;
 	while(<SHADOW>) {
 		chop;
 		($shadowUser) = split(/:/, $_);
